### PR TITLE
fix: stop polling in device challenge view

### PR DESCRIPTION
**Pull request by jhe-okta**
once polling responds with error, also, remove spinner

OKTA-278071

authenticator-siw-poll-fail-stop

### CODE REVIEW

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 1]
```diff
@@ -13,127 +14,122 @@ const request = (opts) => {
...
+      this.startDevicePolling();
+    },
+
+    onPollingFail () {
+      this.$('.spinner').hide();
+      this.stopPolling();
+    },
+
+    remove () {
+      BaseForm.prototype.remove.apply(this, arguments);
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: when does this get called? When device polling succeeds and you render a new screen?

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 2]
```diff
@@ -13,127 +14,122 @@ const request = (opts) => {
...
+        foundPort = true;
+        return request({
+          url: getAuthenticatorUrl('challenge'),
+          method: 'POST',
+          data: JSON.stringify({ challengeRequest }),
+          timeout: 3000 // authenticator should respond within 3000ms for challenge request
+        });
+      };
+
+      const onFailure = () => {};
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: why do we have this empty onFailure method. Can we at least log something use in here ?

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 3]
```diff
@@ -13,127 +14,122 @@ const request = (opts) => {
...
+
+    initialize () {
+      BaseForm.prototype.initialize.apply(this, arguments);
+      this.listenTo(this.model, 'error', this.onPollingFail);
+      this.deviceChallengePollRemediation = this.options.appState.getCurrentViewState();
+      this.doChallenge();
+      this.startDevicePolling();
+    },
+
+    onPollingFail () {
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: 👍

File: src/v2/view-builder/views/DeviceChallengePollView.js (hunk 1 above)
Comment (jhe-okta) [reply src/v2/view-builder/views/DeviceChallengePollView.js]: Yes.
This actually needs improvement as I talked to Haisheng before. Currently, SIW is being too aggressively in rerendering the view, it simply compares the before and after IDX response object, as long as there's any diff, model triggers change event thus view rerender. This does not work well at least in polling view. Sometimes the only difference in responses between pollings is only the timestamp, we don't want the view to refresh in that case.

File: src/v2/view-builder/views/DeviceChallengePollView.js (hunk 2 above)
Comment (jhe-okta) [reply src/v2/view-builder/views/DeviceChallengePollView.js]: I can add some log, but practically it is never going to get there.

File: src/v2/view-builder/views/DeviceChallengePollView.js [hunk 4]
```diff
@@ -13,127 +14,124 @@ const request = (opts) => {
...
+        return request({
+          url: getAuthenticatorUrl('challenge'),
+          method: 'POST',
+          data: JSON.stringify({ challengeRequest }),
+          timeout: 3000 // authenticator should respond within 3000ms for challenge request
+        });
+      };
+
+      const onFailure = () => {
+        Logger.error(`Something unexpected happens on checking port ${currentPort}.`);
```
Comment (nikhilvenkatraman-okta) [inline src/v2/view-builder/views/DeviceChallengePollView.js]: nit: `Something unexpected happened while we were checking port ${currentPort}`

**Pull request merged**